mongodbatlas/resource_mongodbatlas_data_lake.go

```diff
@@ -0,0 +1,484 @@
+		return fmt.Errorf(errorDataLakeSetting, "aws_external_id", name, err)
+	}
+
+	if err := d.Set("data_process_region", flattenDataLakeProcessRegion(&dataLake.DataProcessRegion)); err != nil {
+		return fmt.Errorf(errorDataLakeSetting, "data_process_region", name, err)
+	}
+
+	if err := d.Set("hostnames", dataLake.Hostnames); err != nil {
+		return fmt.Errorf(errorDataLakeSetting, "hostnames", name, err)
+	}
+
+	if err := d.Set("state", dataLake.State); err != nil {
+		return fmt.Errorf(errorDataLakeSetting, "state", name, err)
+	}
+
+	if err := d.Set("storage_databases", flattenDataLakeStorageDatabases(dataLake.Storage.Databases)); err != nil {
+		return fmt.Errorf(errorDataLakeSetting, "storage_databases", name, err)
+	}
+
+	if err := d.Set("storage_stores", flattenDataLakeStorageStores(dataLake.Storage.Stores)); err != nil {
+		return fmt.Errorf(errorDataLakeSetting, "storage_stores", name, err)
+	}
+
+	d.SetId(encodeStateID(map[string]string{
+		"project_id": projectID,
+		"name":       name,
+	}))
+
+	return nil
+}
+
+func resourceMongoDBAtlasDataLakeUpdate(d *schema.ResourceData, meta interface{}) error {
+	// Get client connection.
+	conn := meta.(*matlas.Client)
+	ids := decodeStateID(d.Id())
+	projectID := ids["project_id"]
+	name := ids["name"]
+
+	dataProcess := &matlas.DataProcessRegion{}
+	awsConfig := matlas.AwsCloudProviderConfig{}
+
+	if d.HasChange("aws_role_id") {
+		awsConfig.RoleID = cast.ToString(d.Get("aws_role_id"))
+	}
+
+	if d.HasChange("aws_test_s3_bucket") {
+		awsConfig.TestS3Bucket = cast.ToString(d.Get("aws_test_s3_bucket"))
+	}
+
+	if d.HasChange("data_process_region") {
+		dataProcess = expandDataLakeDataProcessRegion(d)
+	}
+
+	dataLakeReq := &matlas.DataLakeUpdateRequest{
+		CloudProviderConfig: &matlas.CloudProviderConfig{AWSConfig: awsConfig},
+		DataProcessRegion:   dataProcess,
+	}
+	_, _, err := conn.DataLakes.Update(context.Background(), projectID, name, dataLakeReq)
+	if err != nil {
+		return fmt.Errorf(errorDataLakeUpdate, name, err)
+	}
+
+	return resourceMongoDBAtlasDataLakeRead(d, meta)
+}
+
+func resourceMongoDBAtlasDataLakeDelete(d *schema.ResourceData, meta interface{}) error {
+	// Get client connection.
+	conn := meta.(*matlas.Client)
+	ids := decodeStateID(d.Id())
+	projectID := ids["project_id"]
+	name := ids["name"]
+
+	_, err := conn.DataLakes.Delete(context.Background(), projectID, name)
+	if err != nil {
+		return fmt.Errorf(errorDataLakeDelete, name, err)
+	}
+
+	return nil
+}
+
+func resourceMongoDBAtlasDataLakeImportState(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	conn := meta.(*matlas.Client)
+
+	projectID, name, s3Bucket, err := splitDataLakeImportID(d.Id())
+	if err != nil {
+		return nil, err
+	}
+
+	u, _, err := conn.DataLakes.Get(context.Background(), projectID, name)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't import user(%s) in data lake(%s), error: %s", name, projectID, err)
```

Comment:
Maybe should be, if I'm reading this right - it's the data lake name that the error would return:
couldn't import data lake (%s) for project (%s), error: %s", name, projectID, err